File: node/src/resources/deliveries.ts
import { TransmitClient } from '../client';
import { ApiDeliveryRequest, ApiDeliveryResponse, FareQuoteRequest } from '../types';

export class Deliveries {
  private client: TransmitClient;

  constructor(client: TransmitClient) {
    this.client = client;
  }

  /**
   * Create a new delivery via API
   */
  async create(data: ApiDeliveryRequest): Promise<ApiDeliveryResponse> {
    return this.client.post<ApiDeliveryResponse>('/api/v1/api-deliveries', data);
  }

  /**
   * List your API deliveries
   */
  async list(params?: Record<string, any>): Promise<ApiDeliveryResponse[]> {
    return this.client.get<ApiDeliveryResponse[]>('/api/v1/api-deliveries', params);
  }

  /**
   * Get delivery details by ID
   */
  async retrieve(deliveryId: string): Promise<ApiDeliveryResponse> {
    return this.client.get<ApiDeliveryResponse>(`/api/v1/api-deliveries/${deliveryId}`);
  }

  /**
   * Get delivery fare quote (does not require authentication necessarily)
   */
  async quote(data: FareQuoteRequest): Promise<any> {
    return this.client.post<any>('/api/v1/api-deliveries/quote', data);
  }
}

File: rust/src/resources/deliveries.rs
use crate::client::{Client, TransmitError};
use crate::types::{ApiDeliveryRequest, FareQuoteRequest};
use serde_json::Value;

pub struct Deliveries<'a> {
    client: &'a Client,
}

impl<'a> Deliveries<'a> {
    pub fn new(client: &'a Client) -> Self {
        Self { client }
    }

    pub async fn create(&self, req: &ApiDeliveryRequest) -> Result<Value, TransmitError> {
        self.client.post("/api/v1/api-deliveries", req).await
    }

    pub async fn list(&self) -> Result<Vec<Value>, TransmitError> {
        self.client.get("/api/v1/api-deliveries").await
    }

    pub async fn retrieve(&self, id: &str) -> Result<Value, TransmitError> {
        let path = format!("/api/v1/api-deliveries/{}", id);
        self.client.get(&path).await
    }

    pub async fn quote(&self, req: &FareQuoteRequest) -> Result<Value, TransmitError> {
        self.client.post("/api/v1/api-deliveries/quote", req).await
    }
}
